src/utils/renderHelpers.ts

```diff
@@ -15,6 +15,7 @@ export interface RenderOptions {
 	namespaceToStringTag: boolean;
 	pluginDriver: PluginDriver;
 	snippets: GenerateCodeSnippets;
+	useOriginalName?: (variable: Variable) => boolean;
```

Comment:
That is a really minimal solution, I like that. Very minor point: For performance reasons, it is better if objects passed to functions always have the same properties. Then the compiler can produce more optimized code. Thus, a better type would be

```js
useOriginalName: ((variable: Variable) => boolean) | null
```

and you add an explicit `null` where the RenderOptions are created.

Reply:
> For performance reasons, it is better if objects passed to functions always have the same properties. Then the compiler can produce more optimized code

Oh, I don' know this, could you give me some cases that can demonstrate this?

Reply:
Ok, it is a technical thing, but if you want to do some reading, I found this: https://erdem.pl/2019/08/v-8-function-optimization#function-states

Basically if you pass objects of different shapes to a function, that makes the function "polymorphic", which means that the generated machine code first needs to determine the shape of the object and then call the correct branch. If all objects passed to a function have the same shape, the function can be monomorphic and will be much faster.

Reply:
It is admittedly a very low-level optimization, but it is also cheap to fix.

Reply:
Ok, I get it, Thanks very much!